Edit the GUI clients guide

ptgott opened the pull request:
Closes #52850

Recommend the authenticated local tunnel approach. To do so:
- Add an introductory paragraph that briefly compares three approaches to configuring a GUI client to access a database: the authenticated tunnel, local proxy, and remote host/port.
- Indicate that we recommend that authenticated tunnel.
- Separate the H3 section that describes `tsh proxy db` into two H3s, one related to the local tunnel and one the standard `tsh proxy db` command.
- Put the section related to the authenticated tunnel first to emphasize it.

ptgott committed:
Edit the GUI clients guide

Closes #52850

Recommend the authenticated local tunnel approach. To do so:
- Add an introductory paragraph that briefly compares three approaches
  to configuring a GUI client to access a database: the authenticated
  tunnel, local proxy, and remote host/port.
- Indicate that we recommend that authenticated tunnel.
- Separate the H3 section that describes `tsh proxy db` into two H3s,
  one related to the local tunnel and one the standard `tsh proxy db`
  command.
- Put the section related to the authenticated tunnel first to emphasize
  it.

ravicious commented on docs/pages/connect-your-client/gui-clients.mdx:
```diff
@@ -28,57 +26,93 @@ work with Teleport.
+  Teleport user's client certificate and checking the certificate issued by the
+  Teleport-protected database. The connection remains authenticated, and clients
+  only need to forward database protocol messages to it.
+- **Unauthenticated local proxy:** The local GUI client needs to load the
```
```suggestion
- **Unauthenticated local proxy:** The GUI client needs to load the
```

I think that adjective might be unnecessary since there are no non-local GUI clients involved.

ravicious commented on docs/pages/connect-your-client/gui-clients.mdx:
```diff
@@ -28,57 +26,93 @@ work with Teleport.
+check the database server certificate issued by a Teleport-protected database.
+There are three ways to do this:
+
+- **Authenticated tunnel (recommended):** The local proxy handles loading the
+  Teleport user's client certificate and checking the certificate issued by the
+  Teleport-protected database. The connection remains authenticated, and clients
+  only need to forward database protocol messages to it.
```
What do you think about mentioning that the local proxy is started by tsh or Teleport Connect? If I were a regular end user, I think it'd be hard for me to understand where that "local proxy" came from here.

ravicious commented on docs/pages/connect-your-client/gui-clients.mdx:
```diff
@@ -28,57 +26,93 @@ work with Teleport.
-client will be connecting to. This command requires that you use Teleport
-Enterprise (Cloud) or, if self-hosting Teleport, have enabled [TLS
-routing](../admin-guides/management/operations/tls-routing.mdx) mode.
+### Authenticated tunnel
```
What do you think about including a link to https://goteleport.com/docs/connect-your-client/teleport-connect/#connecting-to-a-database in this section? Teleport Connect is not mentioned in this page at all, but at the moment databases are the most commonly used resources in Connect.

ptgott committed:
Respond to ravicious feedback

- Remove unnecessary adjective.
- Clarify that `tsh` and Teleport Connect start the local proxy.
- Mention Teleport Connect in the intro section, since in-app
  instructions provide the function of this guide.

ptgott commented:
@mmcallister Checking if you'd have some time to review this one. Thanks!

ravicious commented on docs/pages/connect-your-client/gui-clients.mdx:
```diff
@@ -37,21 +41,22 @@ GUI clients need to present a certificate to the Teleport Database Service and
+  checking the certificate issued by the Teleport-protected database. The
+  connection remains authenticated, and clients only need to forward database
+  protocol messages to it.
+- **Unauthenticated local proxy:** `tsh` or Teleport Connect starts a local
+  proxy for database traffic, including authentication from clients. The GUI
+  client needs to load the Teleport user's certificate and check database
+  certificates. This requires configuring the client to access TLS credentials,
+  including the user certificate and private key, and a certificate issued by
+  the database's certificate authority. 
```
"check database certificates" in this context is confusing for me because I don't think there's any checking involved on the side of the GUI client. It rather needs to provide these certificates for the connection.

I think it could be simplified to something like this:

> tsh starts a local proxy which routes connections to the Teleport-protected database, but the proxy itself doesn't handle authentication. This requires configuring the database GUI client to provide the user certificate and the private key, as well as a CA certificate, all available under the paths reported by `tsh proxy db`.

ptgott replied:
Thanks! Added this wording with some minor edits to tighten it up and avoid mentioning `tsh proxy db` before we introduce the command.

ptgott committed:
Respond to ravicious feedback

- Simplify explanations of the authenticated/unauthenticated proxy
  approaches.
- Indicate that Teleport Connect only supports the authenticated proxy.

ravicious commented:
@greedy52 Do you want to take a look at the updates here?
